server/src/gnome/prompter.rs

```diff
@@ -0,0 +1,287 @@
+// org.gnome.keyring.Prompter
+// https://gitlab.gnome.org/GNOME/gcr/-/blob/main/gcr/org.gnome.keyring.Prompter.xml
+
+use clap::error::Result;
+use oo7::dbus::ServiceError;
+use serde::{Deserialize, Serialize};
+use zbus::{
+    interface, proxy,
+    zvariant::{DeserializeDict, OwnedObjectPath, OwnedValue, SerializeDict, Type, Value},
+};
+
+use crate::{
+    gnome::secret_exchange::SecretExchange,
+    prompt::{Prompt, PromptRole},
+    service::Service,
+};
+
+#[derive(Debug, DeserializeDict, SerializeDict, Type)]
+#[zvariant(signature = "a{sv}")]
+// System prompt properties
+pub struct Properties {
+    title: Option<String>,
+    #[zvariant(rename = "choice-label")]
+    choice_label: Option<String>,
+    description: Option<String>,
+    message: Option<String>,
+    #[zvariant(rename = "caller-window")]
+    caller_window: Option<String>,
+    warning: Option<String>,
+    #[zvariant(rename = "password-new")]
+    password_new: Option<bool>,
+    #[zvariant(rename = "password-strength")]
+    password_strength: Option<u32>,
+    #[zvariant(rename = "choice-chosen")]
+    choice_chosen: Option<bool>,
+    #[zvariant(rename = "continue-label")]
+    continue_label: Option<String>,
+    #[zvariant(rename = "cancel-label")]
+    cancel_label: Option<String>,
+}
+
+#[derive(Debug, Type)]
+#[zvariant(signature = "s")]
+// Possible values for PromptReady reply parameter
+pub enum Reply {
+    No,
+    Yes,
+    Empty,
```

Comment:
What does empty corresponds to? Please link in the code to where those answers are defined

Reply:
Empty corresponds to: `''`. The first PromptReady call's `reply` argument is always `''` and also the `properties` dict is empty.

![image](https://github.com/user-attachments/assets/e0c12bcd-c794-43b2-bc4e-d643d518ac68)

Reply:
Check if you can use the zvariant::Optional type here and drop the Empty variant.

Reply:
Will do, thanks!

Reply:
hey. I tried `reply: Optional<Reply>` approach and I think the current approach with Empty variant is bit better or shorter. Because all we have to do is impl Serialize and Deserialize. When using Optional, I had to impl `NoneValue` and bunch of String, &str traits. I just stopped at: trait `FromStr` is not implemented for `Reply`. Should I keep `Empty` variant or continue with `Optional`?

Reply:
Use zvariant::Optional please. and you shouldn't have to implement that many traits but if you are struggling with it, you can leave it as is and i will take care of it before merging.

Reply:
Ok, thanks :)

Reply:
Done.